src/core/auth.ts

```diff
@@ -8,15 +8,12 @@ import { saleorAuthToken } from "./constants";
 import { LOGIN, REGISTER } from "../apollo/mutations";
 import { USER } from "../apollo/queries";
 import { loginOpts, registerOpts } from "./types";
+import { Mutation } from "../apollo/types";
 
 export interface AuthSDK {
-  login: (
-    opts: loginOpts
-  ) => Promise<FetchResult<any, Record<string, any>, Record<string, any>>>;
-  logout: () => Promise<ApolloQueryResult<any>[] | null>;
-  register: (
-    opts: registerOpts
-  ) => Promise<FetchResult<any, Record<string, any>, Record<string, any>>>;
+  login: (opts: loginOpts) => Promise<FetchResult<Mutation>>;
```

Comment:
The promise return is too generic, we need to type it more strictly. Codegen should generate exact response types for each promise result